system/jlib/jthread.cpp

```diff
@@ -2005,16 +2005,67 @@ protected: friend class PipeWriterThread;
         return argv;
     }
 
+    bool isPlatformProcess(HANDLE pid)
+    {
+        // We confirm that a child process being executed is one of the platform executables by
+        // checking that its executable location is the same as the current executable
+        // If you ask too fast you get a false positive!
+        try
+        {
+            StringBuffer childPath;
+            getPidExecutablePath(pid, childPath);
+            const char* me = queryCurrentProcessPath();
+            if (streq(me, childPath))
+            {
+                // Probably asked too fast...
```

Comment:
A for/while loop with a max wait of a few seconds ?

Reply:
A few milliseconds should be enough. I could try a few times but not sure it's necessary. The window is between the fork() and the exec() on the child pid...

Reply:
I'm sure in 99.999% of the cases 20 ms is enough, but thought it was a decent general suggestion.

Reply:
Yes, I wondered if I should loop too. This code is designed to illustrate a potential approach rather than be the final version (which will probably need to be made more robust in various ways).